zipline/gens/sim_engine.pyx

```diff
@@ -39,36 +38,39 @@ cdef class MinuteSimulationClock:
                  trading_days,
                  market_opens,
                  market_closes,
-                 all_trading_days,
                  minute_emission=False):
         self.minute_emission = minute_emission
         self.market_opens = market_opens
         self.market_closes = market_closes
         self.trading_days = trading_days
-        self.all_trading_days = all_trading_days
         self.minutes_by_day = self.calc_minutes_by_day()
 
     @cython.boundscheck(False)
     @cython.wraparound(False)
-    cpdef market_minutes(self, np.intp_t i):
+    cdef np.ndarray[np.int64_t, ndim=1] market_minutes(self, np.intp_t i):
```

Comment:
@ehebert I don't have a strong feeling either way, though looking at that stack trace, it looks like Pandas is treating the memoryview you're returning as a scalar.  `to_datetime` is "smart" and will work on all manner of different data structures (arrays, lists, strings, ints, etc).  Since we know what we have (an array of ints), and what we want returned (a DatetimeIndex), I wonder if there's a more "direct" constructor we could use.